File: d1/part1.py
with open('input.txt') as input_file:
    lines = input_file.readlines()

result = 0
for line in lines:
    first_num = ''
    for char in line:
        if char in ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9']:
            first_num = char
    second_num = ''
    for char in line[::-1]:
        if char in ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9']:
            second_num = char
    result += int(second_num+first_num)
print(result)

File: d06/part1.rs
fn main() {
    let input = read_input_from_file("input.txt");
    let times = input[0]
        .split_whitespace()
        .skip(1)
        .map(|s| s.parse::<u32>().unwrap())
        .collect::<Vec<u32>>();
    let distances = input[1]
        .split_whitespace()
        .skip(1)
        .map(|s| s.parse::<u32>().unwrap())
        .collect::<Vec<u32>>();

    let mut result = 1;
    for i in 0..times.len() {
        let mut win = 0;
        for time in 0..times[i] {
            if time * (times[i] - time) > distances[i] {
                win += 1;
            }
        }
        result *= win;
    }
    println!("Result {}", result);
}

fn read_input_from_file(file_path: &str) -> Vec<String> {
    std::fs::read_to_string(file_path)
        .unwrap()
        .lines()
        .map(String::from)
        .collect()
}
